flask_appbuilder/security/decorators.py

```diff
@@ -27,7 +27,7 @@ def wraps(self, *args, **kwargs):
         else:
             log.warning(LOGMSG_ERR_SEC_ACCESS_DENIED.format(permission_str, self.__class__.__name__))
             flash(as_unicode(FLAMSG_ERR_SEC_ACCESS_DENIED), "danger")
-        return redirect(url_for(self.appbuilder.sm.auth_view.__class__.__name__ + ".login"))
+        return redirect(url_for(self.appbuilder.sm.auth_view.__class__.__name__ + ".login", next=request.url))
```

Comment:
I see this fix for only AuthOAuthView, but how do ensure this `next=request.url` variable will be utilized in AuthDBView, AuthLDAPView, 
AuthRemoteUserView and they are sent to the index page, instead of the page they tried to visit initially.

Reply:
True @ankursinghal2005, One would have to replicate the same logic to all child classes.

Hey @betodealmeida are willing to do it?

Reply:
I'm not the right person to do it, since I don't have access to an LDAP auth server. Maybe @ankursinghal2005 can do it for the auth system they're using?

Reply:
@betodealmeida 
I have tested below LDAP test server. It is working fine. You can see it for your reference. I can further work on remaining child classes after this PR gets merged into master.

**Online LDAP Test Server Link**
https://www.forumsys.com/tutorials/integration-how-to/ldap/online-ldap-test-server/
**Configuration -:** 
### When using LDAP Auth, setup the ldap server
AUTH_TYPE = AUTH_LDAP
AUTH_LDAP_SERVER = "ldap://ldap.forumsys.com:389"
AUTH_LDAP_SEARCH = "dc=example,dc=com"
AUTH_LDAP_BIND_USER = "uid=riemann,dc=example,dc=com"
AUTH_LDAP_BIND_PASSWORD = "password"

Reply:
nice!!